File: easy/array/Check if Number Has Equal Digit Count and Digit Value.cpp
class Solution {
public:
    bool digitCount(string num) {
        map<int,int>idxCount , numCount;

        for(int i=0;i<num.size();i++){
            numCount[num[i] - '0']++;
            if(num[i] != '0') idxCount[i] = (num[i] - '0');
        }return idxCount == numCount;
    }
};

File: easy/array/Check if Number Has Equal Digit Count and Digit Value.cs
public class Solution {
    public bool DigitCount(string num) {
        int[] count = new int[10];

        foreach (char c in num) {
            count[c - '0']++;
        }

        for (int i = 0; i < num.Length; i++) {
            int expected = num[i] - '0';
            if (count[i] != expected) {
                return false;
            }
        }

        return true;
    }
}
